app/services/crown_court_outcome_creator.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+class CrownCourtOutcomeCreator < ApplicationService
+  GUILTY_VERDICTS = %w[GUILTY_BUT_OF_LESSER_OFFENCE_BY_JURY_CONVICTED
+                       GUILTY_CONVICTED
+                       GUILTY_BY_JURY_CONVICTED
+                       GUILTY_BUT_OF_LESSER_OFFENCE_CONVICTED].freeze
+
+  def initialize(defendant:)
+    @defendant = defendant
+  end
+
+  def call
+    return unless result_is_a_conclusion?
+
+    {
+      ccooOutcome: trial_outcome,
+      caseEndDate: case_end_date,
+      appealType: nil
+    }
+  end
+
+  private
+
+  def result_is_a_conclusion?
+    defendant&.dig(:offences)&.any? { |offence| offence.dig(:verdict).present? }
```

Comment:
Since defendant is a required param, I'm wondering if perhaps we could live without the safe navigation on `defendant.dig`?
```suggestion
    defendant.dig(:offences)&.any? { |offence| offence.dig(:verdict).present? }
```

Reply:
Good point, fixed 👍